src/NodeHandler.cc

```diff
@@ -3,57 +3,124 @@
 #include "ObjectProxyFactory.h"
 #include "CallbackHandler.h"
 #include "NodeApplication.h"
+#include "TemplateFactory.h"
 
 #include <TROOT.h>
 #include <string>
 
 namespace rootJS {
 
-	NodeHandler *NodeHandler::instance;
-	bool NodeHandler::initialized;
-
-	void NodeHandler::initialize(v8::Local<v8::Object> exports, v8::Local<v8::Object> module) {
-		if(!initialized) {
-			NodeApplication::CreateNodeApplication();
-			ObjectProxyFactory::initializeProxyMap();
-			instance = new NodeHandler(exports);
-			instance->exposeROOT();
-		} else {
-			v8::Isolate::GetCurrent()->ThrowException(v8::String::NewFromUtf8(v8::Isolate::GetCurrent(), "The NodeHandler can only be initialized once."));
+NodeHandler *NodeHandler::instance;
+bool NodeHandler::initialized;
+
+void NodeHandler::exposeGlobalFunctions() {
+}
+
+void NodeHandler::exposeMacros() {
+}
+
+void NodeHandler::exposeClasses() {
+	TCollection* classes = gROOT->GetListOfClasses();
+	TIter next(classes);
+	while (TObject *clazz = next()) {
+		if (((TClass*) clazz)->Property() & kIsClass) {
+			std::string splitname = (clazz->GetName());
+			std::stack stk = splitClassName(splitname);
+			std::string curname = stk.top();
+			this->exports->Set(curname.c_str(),
+					exposeClassRec(stk.top(), stk)->GetFunction());
 		}
 	}
+}
 
-	NodeHandler::NodeHandler(v8::Local<v8::Object> exports) {
-		this->exports = exports;
+v8::Local<v8::FunctionTemplate> NodeHandler::exposeClassRec(std::string name,
+		std::stack<std::string>& stk) {
+	std::string curname = name;
+	stk.pop();
+	DictFuncPtr_t dictFunc = gClassTable->GetDict(name.c_str());
+	//TODO implement map to avoid duplicates and check for double assignments of identical
+	// classes to the same class/name space and avoid them
+	if (dictFunc == nullptr) {
+		//TODO throw exception
 	}
+	if (stk.empty()) {
+		return TemplateFactory::createTemplate(TClassRef(dictFunc()));
+	} else {
+		v8::Local<v8::FunctionTemplate> rectmpl = exposeClassRec(stk.top(),
+				stk);
+		v8::Local<v8::FunctionTemplate> curtmpl =
+				TemplateFactory::createTemplate(TClassRef(dictFunc()));
+		curtmpl->Set(v8::Isolate::GetCurrent(), curname.c_str(), rectmpl);
+		return curtmpl;
+	}
+}
+
+std::stack<std::string> NodeHandler::splitClassName(std::string name) {
 
-	void NodeHandler::exposeROOT() {
-		exposeGlobals();
+	std::string delimiter = "::";
+	std::stack<std::string> stk;
+	std::stack<std::string> revstk;
+	size_t pos = 0;
+	std::string token;
+	while ((pos = name.find(delimiter)) != std::string::npos) {
+		token = name.substr(0, pos);
+		stk.push(token);
+		name.erase(0, pos + delimiter.length());
```

Comment:
Not sure, but you are adding the first part of the string to the stack first, so when you have 
namespace::class your stack looks like class -> namespace.
So when you pop the stack you get class but class needs to be attached to the namespace v8-Object which is not there yet because it will be exposed with the next stack pop...